include/wlr/types/wlr_foreign_toplevel_management_v1.h

```diff
@@ -0,0 +1,112 @@
+/*
+ * This an unstable interface of wlroots. No guarantees are made regarding the
+ * future consistency of this API.
+ */
+#ifndef WLR_USE_UNSTABLE
+#error "Add -DWLR_USE_UNSTABLE to enable unstable wlroots features"
+#endif
+
+#ifndef WLR_TYPES_WLR_FOREIGN_TOPLEVEL_MANAGEMENT_V1_H
+#define WLR_TYPES_WLR_FOREIGN_TOPLEVEL_MANAGEMENT_V1_H
+
+#include <wayland-server.h>
+#include <wlr/types/wlr_output.h>
+
+struct wlr_foreign_toplevel_manager_v1 {
+	struct wl_global *global;
+	struct wl_list resources;
+	struct wl_list toplevels; // wlr_foreign_toplevel_handle_v1::link
+
+	struct wl_listener display_destroy;
+
+	struct {
+		struct wl_signal destroyed;
+	} events;
+
+	void *data;
+};
+
+enum wlr_foreign_toplevel_handle_state {
+	WLR_FOREIGN_TOPLEVEL_STATE_MAXIMIZED = 1,
+	WLR_FOREIGN_TOPLEVEL_STATE_MINIMIZED = 2,
+	WLR_FOREIGN_TOPLEVEL_STATE_ACTIVATED = 4,
+};
+
+struct wlr_foreign_toplevel_handle_output {
+	struct wl_list link; // wlr_foreign_toplevel_handle_v1::outputs
+	struct wlr_output *output;
+};
+
+struct wlr_foreign_toplevel_handle_v1 {
+	struct wl_list resources;
+	struct wl_list link;
+
+	char *title;
+	char *app_id;
+	struct wl_list outputs; // wlr_foreign_toplevel_output
```

Comment:
I'd like not to have a list of outputs here. We should add it to `wlr_surface` later instead.

(Additionally, this doesn't handle outputs being destroyed)

Reply:
Yep, we need some way to track this, and except if we have some list of outputs in wlr_surface, I see no way around it. But even then we will need to associate a toplevel with a surface, which the current implementation avoids (not sure if that brings any value though). It could be a follow-up issue, but I'll surely update it to handle output destruction.

Reply:
Okay, that seems reasonable.